torch/_meta_registrations.py

```diff
@@ -283,6 +283,11 @@ def meta_fft_c2r(self, dim, normalization, lastdim):
 
 @register_meta(aten.copy_.default)
 def meta_copy_(self, src, non_blocking=False):
+    # Manually running copy_() in the meta function here,
+    # to ensure we run the same assertions that the eager
+    # copy_() kernel would ordinarily run.
+    self_clone = utils.clone_preserve_strides(self)
+    self_clone.copy_(src)
```

Comment:
@ezyang does this meta function look ok to you? Replicating all the error checking in our `copy_()` kernel is pretty annoying - I figure it's pretty harmless to do this since we're running with meta tensors.

Reply:
nvm this doesn't work (when I wrote it I thought I was writing the meta impl for `aten.copy`, not `aten.copy_`. But this will just infinite loop).

Hmm, maybe we'll need to replicate some of the meta checks in the C++ `copy_()` impl (or figure out a way to call it here - since it's `CompositeExplicitAutograd`)